src/groupeddataframe/utils.jl

```diff
@@ -303,6 +304,31 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
     else
         sorted = false
     end
+
+    lg = length(groups)
+    nt = Threads.nthreads()
+    # disable threading if we are processing a small data frame or number of groups is large
+    if lg < 1_000_000 || ngroups > lg * (0.5 - 1 / (2 * nt)) / (2 * nt)
+       nt = 1
+    end
+    seen = fill(false, ngroups)
+    seen_vec = Vector{Vector{Bool}}(undef, nt)
+    seen_vec[1] = seen
+    for i in 2:nt
+        seen_vec[i] = fill(false, ngroups)
+    end
+
+    # this should be very cheap to compute as nt is not large
+    border_chunks = [0; (lg .÷ nt) .* (1:nt-1) ; lg]
```

Comment:
Could we reuse `split_indices` for this?

Reply:
no - split indices works the other way around. There you give the `basesize` and here you give number of chunks (actually this case is simpler conceptually).

Reply:
Ah - now I look at `(Int(1 + ((i - 1) * len′) ÷ np):Int((i * len′) ÷ np) for i in 1:np)` and this is the same.

Reply:
done